### ISSUE TITLE
Allow setting selected row on VirtualizedList (FlatList/SectionList) by clicking a row

### ISSUE THREAD

**Issue by lyahdav** [microsoft/react-native-macos#539]
# Proposal: Allow setting selected row on VirtualizedList (FlatList/SectionList) by clicking a row

## Summary
Currently `VirtualizedList` has a `selectedRowIndex` in its state that gets set when pressing up/down arrow keys if you set the `enableSelectionOnKeyPress` prop to `true`. But typically you can select a selected row in a list component also by clicking on a row. Right now `VirtualizedList` doesn't expose a way to do so.

## Motivation

In Summary.

## Basic example

Perhaps it would be sufficient to add a `selectedRowIndex` prop?

## Open Questions

Possibly as part of [this PR](https://github.com/microsoft/react-native-windows/pull/5590) this should get upstreamed to RN core.

**Comment by chrisglein** [comment]
@lyahdav I assume this is behavior you want for all keyboard platforms, not just macOS, is that correct?
From the issue template we have some guidance on that:
> - For feature requests to React Native, go [here](https://github.com/react-native-community/discussions-and-proposals). 
> - For feature requests to React Native for Windows (or) Desktop features in general, go [here](https://github.com/microsoft/react-native-windows/issues/new?assignees=&labels=Proposal&template=react-native-windows--vnext--feature-request.md&title=Your+feature+request).

Just want to confirm this is something we also want in Windows. If so I can transfer this issue to that repo or create a tracking issue as we'll need to do work in both.

**Comment by lyahdav** [comment]
Sorry for delay here. We ended up not using the VirtualizedList row selection features for now so this isn't a priority. I think it makes sense for this to be implemented for all platforms eventually.

**Comment by Saadnajmi** [comment]
Heh. I ended up implementing this recently with a 'selectRowAtIndex' method on the listRef.
